crypto/evp/evp_fetch.c

```diff
@@ -298,7 +299,8 @@ inner_evp_generic_fetch(struct evp_method_data_st *methdata,
         unsupported = 1;
 
     if (meth_id == 0
-        || !ossl_method_store_cache_get(store, meth_id, properties, &method)) {
+        || !ossl_method_store_cache_get(store, prov, meth_id, properties,
```

Comment:
Removing this call to `ossl_method_store_cache_get()` I can get it to work consistently. This call returns 0 (not in cache?) in many cases so the wrong fetch is not always seen. Something still not quite right with fetching from cache?

Reply:
Not having the cache around is going to hit performance *hard*.  We've tested that.

Reply:
It's quite possible, though, that I missed a detail somewhere...  I'll have another look.

Reply:
Ah-hah!  I missed a spot in `ossl_method_store_cache_set`!

Reply:
I just pushed a fixup which might make you happier.  Please try.